hubverse_annotator/app.py

```diff
@@ -86,83 +89,180 @@ def forecast_annotation_ui(
         selected target.
     """
-    models = single_loc_hub_table["model"].unique().sort().to_list()
+    models = (
+        forecast_table.filter(pl.col("location") == loc_abbr)
+        .get_column("model_id")
+        .unique()
+        .sort()
+        .to_list()
+    )
     selected_models = st.multiselect(
-        "Model(s)", options=models, default=models, key="model_selection"
+        "Model(s)",
+        options=models,
+        default=None,
+        key="model_selection",
+    )
+
+    forecast_targets = (
+        forecast_table.filter(
+            pl.col("location") == loc_abbr,
+            pl.col("model_id").is_in(selected_models),
+        )
+        .get_column("target")
+        .unique()
+        .sort()
+        .to_list()
     )
-    targets = (
-        single_loc_hub_table.filter(pl.col("model").is_in(selected_models))
+    observed_data_targets = (
+        observed_data_table.filter(pl.col("location") == loc_abbr)
         .get_column("target")
         .unique()
         .sort()
         .to_list()
     )
+    all_targets = sorted(set(forecast_targets + observed_data_targets))
     selected_target = st.selectbox(
-        "Target(s)", options=targets, key="target_selection"
+        "Target",
+        options=all_targets,
+        key="target_selection",
     )
     return selected_models, selected_target
 
 
+def get_available_locations(
+    observed_data_table: pl.DataFrame, forecast_table: pl.DataFrame
+) -> pl.DataFrame:
+    """
+    Retrieves a dataframe of locations from forecasttools
+    used for converting between location formats. The
+    dataframe is cached for streamlit via cache_data.
+
+    Parameters
+    ----------
+    observed_data_table : pl.DataFrame
+        A hubverse table of loaded data (possibly empty).
+    forecast_table : pl.DataFrame
+        The hubverse formatted table of forecasted ED
+        visits and or hospital admissions (possibly empty).
+
+    Returns
+    -------
+    pl.DataFrame
+        A dataframe of locations in different formats.
+    """
+    locs = (
+        pl.concat(
+            [
+                observed_data_table.get_column("location"),
+                forecast_table.get_column("location"),
+            ]
+        )
+        .unique()
+        .to_list()
+    )
+    return forecasttools.location_lookup(
+        location_vector=list(set(locs)), location_format="abbr"
+    )
+
+
+def get_reference_dates(forecast_table: pl.DataFrame) -> list[str]:
+    """
+    Retrieves a dataframe of forecast reference dates. The
+    dataframe is cached for streamlit via cache_data.
+
+    Parameters
+    ----------
+    forecast_table : pl.DataFrame
+        The hubverse formatted table of forecasted ED
+        visits and or hospital admissions (possibly empty).
+
+    Returns
+    -------
+    list[str]
+        A list of available reference dates.
+    """
+    return forecast_table["reference_date"].unique().to_list()
```

Comment:
I have a style preference for `df.get_column("col")` over `df["col"]` in general but especially in method chains like this one.
```suggestion
    return forecast_table.get_column("reference_date").unique().to_list()
```